fastcrawler/parsers/selectors/css.py

```diff
@@ -0,0 +1,38 @@
+from typing import Any, List
+
+from lxml import etree
+from lxml import html as lxml_html
+
+from fastcrawler.parsers.html import HTMLParser
+from fastcrawler.parsers.pydantic import T
+
+from .base import BaseSelector
+
+
+class _CSSField(BaseSelector):
+    """
+    CSSSelectorField represents a field that can be retrieved from a given HTML
+        document using CSS selectors.
+    """
+    def __init__(
+        self,
+        query: str,
+        extract: str | None = None,
+        many: bool = False,
+        model: T | None = None
+    ):
+        super(_CSSField, self).__init__(query, extract, many, model)
+        self.parser = HTMLParser
+
+    def resolve(self, html: str, model: None | T = None) -> T | list[T | None]:
+        """ Resolves HTML input using CSS selector
+        """
+        self.model = model or self.model
+        tree = lxml_html.fromstring(html)
+        results: List[etree.ElementBase] = tree.cssselect(self.query)
+        res = self._process_results(results)
+        return res
+
+
+def CSSField(**kwargs) -> Any:
```

Comment:
for type hinting, we assign data (Any data) to a css field. it's not declarative here, same goes in pydantic. when you import Field, it actually returns Any, because we don't know what field is going to exactly return, it can be int, string, list, or ANY other stuff.
I might be wrong tho, that was how Depends used in fastapi or Field in pydantic.

Reply:
so why it is a function and we dont use  class directly
@ManiMozaffar

Reply:
Because let's say you do this in your class

``python
from fastcrawler import CSSField, BaseModel

class MyData(BaseModel):
    id: int = CSSField(...)

if CSSField is class, then it fixes for type annonation in __init__, which return none, so id must be none according to type annotation , however we use function to control this behaviour and then we say we return ANY. but in fast we still returned CSSField.

https://github.com/tiangolo/fastapi/blob/master/fastapi/param_functions.py

https://github.com/pydantic/pydantic/blob/main/pydantic/fields.py

actually i should do Depends same as well, to initiate a function from it. kwargs should be passed as keywords, not how i did it here, for keyword ref in IDEs. I'll fix the **kwargs part.

What do you think about function thing? I think we should follow best practices